vm/devices/net/netvsp/src/lib.rs

```diff
@@ -206,6 +207,17 @@ impl<T: RingMem + 'static + Sync> InspectTaskMut<Worker<T>> for NetQueue {
             );
 
             if let WorkerState::Ready(state) = &worker.state {
+                // Sync the coalesced packet count from the underlying queue
+                if let Some(queue_state) = &self.queue_state {
+                    if let Some(current_count) = queue_state.queue.tx_packets_coalesced() {
+                        let last_count = state.state.stats.tx_packets_coalesced.get();
+                        let diff = current_count.saturating_sub(last_count);
+                        if diff > 0 {
+                            state.state.stats.tx_packets_coalesced.add(diff);
+                        }
```

Comment:
I don't think you should be draining the underlying queue counter. I know there's no set() for counters, but you can discard the old value. Example:

state.state.stats.tx_packets_coalesced = SharedCounter::new(); // back to 0
state.state.stats.tx_packets_coalesced.add(current_count);